syntax = "proto3";

service TaskService {
  rpc CreateTask (Task) returns (TaskResponse);
  rpc GetTask (TaskRequest) returns (Task);
  rpc UpdateTask (Task) returns (TaskResponse);
  rpc DeleteTask (TaskRequest) returns (TaskResponse);
  rpc ListTasks (Empty) returns (TaskList);
}

message Task {
  string id = 1;
  string title = 2;
  string status = 3;
}

message TaskRequest {
  string id = 1;
}

message TaskResponse {
  string message = 1;
}

message TaskList {
  repeated Task tasks = 1;
}

message Empty {}